### PR TITLE
Better save

### PR DESCRIPTION
**Pull request by janessalee11**
- When saving content items, widgets, pages, users, tags, etc, it returns to the previous page. 
- Success messages now contain a link back to the edited item.

### CODE REVIEW

File: admin/controllers/categories/views/edit/model.php [hunk 1]
```diff
@@ -62,7 +62,8 @@
 			// forms are valid, save info
 			$saved = $cat->save($required_details_form, $custom_fields_form);
 			if ($saved) {
-				CMS::Instance()->queue_message('Category saved','success',Config::uripath() . "/admin/categories");
+				$msg = 'Category <a href="' . Config::uripath() . '/admin/categories/edit/' . $cat->id . '">' . $cat->title . '</a> ' . ($new_cat ? 'created' : 'updated');
```
Comment (workaj) [inline admin/controllers/categories/views/edit/model.php:65]: is there a reason not to use templating?

File: admin/controllers/content/views/edit/model.php [hunk 2]
```diff
@@ -95,7 +95,7 @@
 			}
 			else {
 				$redirect_to = Config::uripath() . "/admin/content/all/" . $content->content_type;
-				$msg = "Content saved";
+				$msg = 'Content <a href="' . Config::uripath() . '/admin/content/edit/' . $content->id . '/' . $content_type . '">' . $content->title . '</a> ' . ($new_content ? 'created' : 'updated');
```
Comment (workaj) [inline admin/controllers/content/views/edit/model.php:98]: templating?

File: admin/controllers/pages/views/save/model.php [hunk 3]
```diff
@@ -12,6 +12,7 @@
 }
 
 if(Input::getvar("id")==0) {
+	$new_page = true;
```
Comment (workaj) [inline admin/controllers/pages/views/save/model.php:15]: looks like this is always true?

File: admin/controllers/pages/views/save/model.php [hunk 4]
```diff
@@ -56,10 +57,9 @@
 		$data = array( $page->id, $_POST['override_positions'][$n], $_POST['override_positions_widgets'][$n], $page->id, $_POST['override_positions'][$n], $_POST['override_positions_widgets'][$n] );
 		$override_success = DB::exec("insert into page_widget_overrides (page_id, position, widgets) values (?,?,?) on duplicate key update page_id=?, position=?, widgets=?", $data);
 	}
-	
-	
 
-	CMS::Instance()->queue_message('Page created/updated','success',Config::uripath().'/admin/pages');
+	$msg = 'Page <a href="' . Config::uripath() . '/admin/pages/edit/' . $page->id . '/' . $page->content_type . '/' . $page->view . '">' . $page->title . '</a> ' . $status . ($new_page ? 'created' : 'updated');
```
Comment (workaj) [inline admin/controllers/pages/views/save/model.php:61]: templating?

File: core/plugin.php [hunk 5]
```diff
@@ -85,7 +85,8 @@ public function save($plugin_options_form) {
 			$result = DB::exec("update plugins set options=? where id=?", [$options_json, $this->id]);
 			
 			if ($result) {
-				CMS::Instance()->queue_message('Plugin options updated','success',Config::uripath() . '/admin/plugins/show');	
+				$msg = 'Plugin <a href="' . Config::uripath() . '/admin/plugins/edit/' . $this->id . '">' . $this->title . '</a> ' . 'updated';
```
Comment (workaj) [inline core/plugin.php:88]: templating?

File: core/widget.php [hunk 6]
```diff
@@ -172,7 +176,7 @@ public function save($required_details_form, $widget_options_form, $position_opt
 			$result = DB::exec("update widgets set state=?, title=?, note=?, options=?, position_control=?, global_position=?, page_list=? where id=?", $params);
 			
 			if ($result) {
-				CMS::Instance()->queue_message('Widget updated','success',Config::uripath() . '/admin/widgets/show');	
+				CMS::Instance()->queue_message('Widget <a href="' . Config::uripath() . '/admin/widgets/edit/'.  $this->id . '">' . $this->title .'</a> updated','success', $redirect_url);
```
Comment (workaj) [inline core/widget.php:179]: templating

File: core/widget.php [hunk 7]
```diff
@@ -182,8 +186,9 @@ public function save($required_details_form, $widget_options_form, $position_opt
 			// new
 			$params = array($this->state, $this->type_id, $this->title, $this->note, $options_json, $this->position_control, $this->global_position, implode(',',$this->page_list)) ;
 			$result = DB::exec("insert into widgets (state,type,title,note,options,position_control,global_position,page_list) values(?,?,?,?,?,?,?,?)", $params);
+			$new_widget_id = DB::get_last_insert_id();
 			if ($result) {
-				CMS::Instance()->queue_message('New widget saved','success',Config::uripath() . '/admin/widgets/show');	
+				CMS::Instance()->queue_message('Widget <a href="' . Config::uripath() . '/admin/widgets/edit/'.  $new_widget_id . '">' . $this->title .'</a> created','success', $redirect_url);
```
Comment (workaj) [inline core/widget.php:191]: templating

File: admin/controllers/categories/views/edit/model.php (hunk 1 above)
Comment (janessalee11) [reply admin/controllers/categories/views/edit/model.php]: fixed in b772c9972b1b6528cd219293d837976ffad29f09

File: admin/controllers/content/views/edit/model.php (hunk 2 above)
Comment (janessalee11) [reply admin/controllers/content/views/edit/model.php]: fixed in b772c9972b1b6528cd219293d837976ffad29f09

File: admin/controllers/pages/views/save/model.php (hunk 4 above)
Comment (janessalee11) [reply admin/controllers/pages/views/save/model.php]: fixed in b772c9972b1b6528cd219293d837976ffad29f09

File: admin/controllers/content/views/edit/model.php (hunk 2 above)
Comment (janessalee11) [reply admin/controllers/content/views/edit/model.php]: fixed in b772c9972b1b6528cd219293d837976ffad29f09

File: core/plugin.php (hunk 5 above)
Comment (janessalee11) [reply core/plugin.php]: fixed in b772c9972b1b6528cd219293d837976ffad29f09

File: core/widget.php (hunk 6 above)
Comment (janessalee11) [reply core/widget.php]: fixed in b772c9972b1b6528cd219293d837976ffad29f09

File: core/widget.php (hunk 7 above)
Comment (janessalee11) [reply core/widget.php]: fixed in b772c9972b1b6528cd219293d837976ffad29f09

**Review by workaj — CHANGES_REQUESTED** [review]
looks good, but `Toggled state of $content` doesnt have links

**Comment by janessalee11** [comment]
### Updates
- Feat: Links added to toggle messages
  image
- Bug: Padding added to toggle button to fix accidental selection of item when trying to toggle.

  Padding before             |  Padding after
  :-------------------------:|:-------------------------:
  image  |  image

**Review by workaj — APPROVED** [review]
seems legit

**Pull request merged**